### PR TITLE
fixup! feat: download wfs files to tmp folder for bucket service

### PR DESCRIPTION
**Pull request by kapil-agnihotri**
Refactor to rename downloaded file to not start with a "." and to have name format as
<datasetId>_<numberOfTotalFeatureProcessed>

ING-3609

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java [hunk 1]
```diff
@@ -458,8 +458,10 @@ private void createNextIterator() {
 				try {
 					URL url = nextUri.toURL();
 					StringBuilder fileName = new StringBuilder(tmpdir);
-					fileName.append("/").append(url.getPath().replaceAll("/", "."))
-							.append(totalFeaturesProcessed).append(".gml");
+
+					fileName.append("/")
+							.append(url.getPath().substring(url.getPath().lastIndexOf("/") + 1))
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java:463]: This works to well identify a service when using a WFS provided by hale connect, but not all WFS will have an identifier as the last element in the path. So if someone wants to use data from multiple services that may be a problem. For different services from the same platform/software it seems to be quite common that the last path element is the same.

Examples of other WFS URLs:

- https://mercator.vlaanderen.be/raadpleegdienstenmercatorpubliek/ps/wfs?SERVICE=WFS&VERSION=2.0.0&REQUEST=GetFeature&NAMESPACES=xmlns%28ps%2Chttps%3A%2F%2Fwww.mercator.vlaanderen.be%2Fps%29&TYPENAMES=ps%3Aps_hbtrl%2Cps%3Aps_vglrl&STARTINDEX=0&COUNT=10
- http://geodienste.bfn.de/arcgis/services/bfn_mns/Benthische_Lebensraeume_wfs/MapServer/WFSServer?SERVICE=WFS&VERSION=1.1.0&REQUEST=GetFeature&NAMESPACE=xmlns%28bfn_mns_Benthische_Lebensraeume_wfs%3Dhttp%3A%2F%2Farcgis0.bfn.cu.ennit.de%3A6080%2Farcgis%2Fservices%2Fbfn_mns%2FBenthische_Lebensraeume_wfs%2FMapServer%2FWFSServer%29&TYPENAME=bfn_mns_Benthische_Lebensraeume_wfs%3ABenthische_Lebensraeume_in_der_deutschen_Ostsee%2Cbfn_mns_Benthische_Lebensraeume_wfs%3ABenthische_Lebensraeume_in_der_deutschen_Nordsee

Comment (kapil-agnihotri) [reply io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java]: Thank you for the feedback. Wasn't aware of additional WFS requests. In this case I think then we can leave the name with `urlPath` and I have modified code to remove first `dot` or `/`.

**Review by stempler — APPROVED** [review]
LGTM

**Comment by stempler** [comment]
@kapil-agnihotri Should this be merged?

**Comment by kapil-agnihotri** [comment]
Yes, I lost the track of this branch.

**Pull request merged**